beacon-chain/state/fieldtrie/field_trie.go

```diff
@@ -222,7 +260,7 @@ func (f *FieldTrie) FieldReference() *stateutil.Reference {
 // Empty checks whether the underlying field trie is
 // empty or not.
 func (f *FieldTrie) Empty() bool {
-	return f == nil || len(f.fieldLayers) == 0
+	return f == nil || len(f.fieldLayers) == 0 || f.isTransferred
```

Comment:
Is `isTransferred` needed at all? Inside `TransferTrie()` you assign `f.fieldLayers = nil`, which means `len(f.fieldLayers) == 0`.